synapse/handlers/appservice.py

```diff
@@ -307,6 +314,23 @@ async def _notify_interested_services_ephemeral(
                             service, "presence", new_token
                         )
 
+                    elif (
+                        stream_key == "to_device_key"
+                        and self.msc2409_to_device_messages_enabled
```

Comment:
should this be an `assert` rather than an `and` ? We should already have checked `msc2409_to_device_messages_enabled`?